services/src/pro/contexts/mod.rs

```diff
@@ -312,6 +393,28 @@ where
     }
 }
 
+pub struct QuotaCheckerImpl<U: UserDb> {
+    user_db: Arc<U>,
+    session: UserSession,
+}
+
+#[async_trait]
+impl<U: UserDb> QuotaCheck for QuotaCheckerImpl<U> {
+    async fn check_quota(&self) -> geoengine_operators::util::Result<bool> {
+        // TODO: cache the result, s.th. other operators in the same workflow can re-use it
+        Ok(self
+            .user_db
+            .quota_available(&self.session)
+            .await
+            .map_err(
+                |e| geoengine_operators::error::Error::QueryingProcessorFailed {
+                    source: Box::new(e),
+                },
+            )?
+            > 0)
+    }
```

Comment:
Do we need to check (or start doing this…) performance degredations?
This way, we could decide to work on this now or… not.

Reply:
I performed a benchmark (see my Trello comment)

>     Starting benchmark, quota check enabled: false
    Run 0 time 249.679598ms
    Run 1 time 242.757376ms
    Run 2 time 243.797694ms
    Run 3 time 247.024981ms

>    Starting benchmark, quota check enabled: true
    Run 0 time 255.951031ms
    Run 1 time 251.298686ms
    Run 2 time 251.200896ms
    Run 3 time 245.754991ms

Reply:
So we leave it there for now?

Reply:
I'd say so